src/module/data/item/ability.mjs

```diff
@@ -228,8 +252,61 @@ export default class AbilityModel extends BaseItemModel {
    * @param {number} [options.banes]  Banes to apply to a power roll
    * @param {number} [options.edges]  Edges to apply to a power roll
    * @returns {Promise<DrawSteelChatMessage>}
+   * TODO: Add hooks based on discussion with module authors
    */
   async use(options = {}) {
+    /**
+     * Configuration information
+     * @type {object | null}
+     */
+    let configuration = null;
+    let resourceSpend = this.resource ?? 0;
+
+    // Determine if the configuration form should even run.
+    // Can be factored out if/when complexity increases
+    if (this.spend.value || this.spend.text) {
+      let content = "";
+
+      /**
+       * Range picker config is ignored by the checkbox element
+       * @type {FormInputConfig} */
+      const spendInputConfig = {
+        name: "spend",
+        min: 0,
+        max: this.actor.system.hero.primary.value,
+        step: 1
+      };
+
+      // Nullish value with text means X spend
+      const spendInput = this.spend.value ?
+        foundry.applications.fields.createCheckboxInput(spendInputConfig) :
+        foundry.applications.elements.HTMLRangePickerElement.create(spendInputConfig);
+
+      content += foundry.applications.fields.createFormGroup({
+        label: game.i18n.format("DRAW_STEEL.Item.Ability.ConfigureUse.SpendLabel", {
+          value: this.spend.value || "",
+          name: this.resourceName
+        }),
+        input: spendInput
+      }).outerHTML;
+
+      configuration = await foundry.applications.api.DialogV2.confirm({
+        content,
+        window: {
+          title: "DRAW_STEEL.Item.Ability.ConfigureUse.Title",
+          icon: "fa-solid fa-gear"
+        },
+        yes: {
+          callback: (event, button, dialog) => {
+            return new FormDataExtended(button.form).object;
+          }
+        },
+        rejectClose: false
+      });
+
+      if (!configuration) throw new Error("Configuration required but not provided");
```

Comment:
If the dialog was cancelled, that should be taken to mean that the user didn't want to use the item after all, no? So it shouldn't throw an error, it should just abort.